client_plugin/utils/powershell/powershell.go

```diff
@@ -32,9 +32,6 @@ import (
 var (
 	// shell is the handle to a single open Powershell session.
 	shell ps.Shell
-
-	// mutex synchronizes access to the single PowerShell session.
```

Comment:
How does this work? Can we say why the mutex was needed in the first place - is it like the powershll. Exec() function can handle only one call at a time. Is it to ensure that request and response are matched and one request (say volume create) doesn't get the response for a parallel volume Get(). This doesn't create a session per volume op. Its creating a session for the plugin instance. And all volume ops that are in flight will now use the single instance without any synchronization.

If we want a session per op then that session has to be created when a volume op request is received.

The init() function is run and the shell global is set when Go run time invokes this package init().

Reply:
As per below information, I guess locking mechanism was placed and ps.Exec() is getting blocked as described at https://github.com/vmware/docker-volume-vsphere/issues/1780#issue-251296227 hence the `120 sec` of threshold was added from https://github.com/vmware/docker-volume-vsphere/pull/1775#discussion_r134000743.

I have removed the dead code as well supplied the correct code, please review it again.

`
"Note that a single shell instance is not safe for concurrent use, as are remote sessions. You can have as many remote sessions using the same shell as you like, but you must execute commands serially. If you need concurrency, you can just spawn multiple PowerShell processes (i.e. call .New() multiple times)."
`

Reference: https://github.com/gorillalabs/go-powershell/blob/master/README.md